drivers/bluetooth/hci/h4.c

```diff
@@ -311,6 +315,17 @@ static inline void read_payload(const struct device *dev)
 
 			LOG_WRN("Failed to allocate, deferring to rx_thread");
 			uart_irq_rx_disable(cfg->uart);
+			/*
+			 * At this time, HCI UART RX is turned off, which means that no new
+			 * received data buffer will be put into the RX FIFO. This will cause
+			 * `rx.ready` to not be modified. It will probably remain unchanged and
+			 * the count of `rx.ready` will probably be 0.
+			 *
+			 * Since it is uncertain whether the RX thread is blocked waiting for
+			 * `rx.ready`, give a semaphore to try to wake up the RX thread.
+			 * Then the RX buffer will be applied again in RX thread.
```

Comment:
"applied again" seems unclear here. I guess you mean "Then there will be a renewed attempt at allocating an RX buffer in the RX thread"?